tests/resolvers/UserTag/usersAssignedTo.spec.ts

```diff
@@ -0,0 +1,100 @@
+import "dotenv/config";
+import { usersAssignedTo as usersAssignedToResolver } from "../../../src/resolvers/UserTag/usersAssignedTo";
+import type {
+  UsersConnectionResult,
+  UserTagUsersAssignedToArgs,
+} from "../../../src/types/generatedGraphQLTypes";
+import { connect, disconnect } from "../../helpers/db";
+import type mongoose from "mongoose";
+import { beforeAll, afterAll, describe, it, expect } from "vitest";
+import type { TestUserTagType } from "../../helpers/tags";
+import type { TestUserType } from "../../helpers/userAndOrg";
+import { createTagsAndAssignToUser } from "../../helpers/tags";
+import { MAXIMUM_FETCH_LIMIT } from "../../../src/constants";
+import { TagUser } from "../../../src/models";
+import { Types } from "mongoose";
+
+let MONGOOSE_INSTANCE: typeof mongoose;
+let testTag: TestUserTagType, testUser: TestUserType;
+
+beforeAll(async () => {
+  MONGOOSE_INSTANCE = await connect();
+  [testUser, , [testTag]] = await createTagsAndAssignToUser();
+});
+
+afterAll(async () => {
+  await disconnect(MONGOOSE_INSTANCE);
+});
+
+describe("resolvers -> Tag -> usersAssignedTo", () => {
+  it(`returns error object when the maximum fetch limit is exceeded`, async () => {
+    const parent = testTag!;
+
+    const args: UserTagUsersAssignedToArgs = {
+      input: {
+        limit: MAXIMUM_FETCH_LIMIT + 1,
+        direction: "FORWARD",
+      },
+    };
+
+    const payload = await usersAssignedToResolver?.(parent, args, {});
+
+    expect(payload!.errors.length).toEqual(1);
+    expect(payload!.errors[0]).toMatchObject({
+      __typename: "MaximumValueError",
+    });
+    expect(payload!.data).toBeNull();
+  });
+
+  it(`returns error object when the cursor provided is invalid`, async () => {
+    const parent = testTag!;
+
+    const args: UserTagUsersAssignedToArgs = {
+      input: {
+        limit: MAXIMUM_FETCH_LIMIT,
+        direction: "FORWARD",
+        cursor: Types.ObjectId().toString(),
+      },
+    };
+
+    const payload = await usersAssignedToResolver?.(parent, args, {});
+
+    expect(payload!.errors.length).toEqual(1);
+    expect(payload!.errors[0]).toMatchObject({
+      __typename: "IncorrectCursor",
+    });
+    expect(payload!.data).toBeNull();
+  });
+
+  it(`returns correct connection object when the arguments are correct`, async () => {
```

Comment:
Have u tested whether forward/backward paginating and default sorting/filtering are working correctly? Doing it for each resolver would be cumbersome, since testing the orm itself would defeat the purpose of using orm. I guess we have to trust the orm on this. Anyway it'd be good to verify all of this is working as expected.

This would require a pre defined list of objects, you have to arrange the objects in that list differently and compare the arrangement with what mognoose query returns for different filter/sorts.

Reply:
@xoldyckk I tested this locally on my system for a couple of user tags and it seemed to work well. Any possible flaws or edge cases can only be highlighted with the adoption of the same i.e. when we add more pagination-based fields and clients begin to use them. 
In such cases, debugging the connection factory would debug all the resolvers too in one short.

Reply:
@EshaanAgg i'm talking about the default filter and sort

Reply:
Yes. Moongoose sorts them in the order they are specified, so just negating them worked well locally.